src/utils.py

```diff
@@ -13,4 +16,24 @@ def calculate_sum(lst):
    """
       
    total = list(map(sum, zip(*lst)))
-   return total
\ No newline at end of file
+   return total
+
+
+
+def write_result(computed_result, file_name):
+    """
+   This function writes result to file
+   Parameters
+   -----------
+   lst: List
+       A list of all the student computed results.
+   str: file name
+       
+   Returns
+   --------
+   None
+   """
+    with open("results/%s.txt" % file_name , 'w') as f:
+        for result in computed_result:
+            for key, value in result.items():
+                f.write('%s:%s\n' % (key, value))
```

Comment:
So this is where it went wrong. 

- First you need to write the output as CSV (see [here](https://www.geeksforgeeks.org/writing-csv-files-in-python/) for examples
- You also need to generate two output CSV files, one for per semester results, another for overall results (CGPA). Please see my previous comments for details and expected file structure